blocklint/main.py

```diff
@@ -38,20 +38,47 @@ def main(args=None):
             with open(file, 'r') as handle:
                 total_issues += process_file(handle, file, word_checkers,
                                              args['end_pos'])
-
     if (args['max_issue_threshold'] is not None
             and args['max_issue_threshold'] <= total_issues):
         print(("Found {issues} issues, with maximum set to "
                "{max}!").format(
                    issues=total_issues,
                    max=args['max_issue_threshold']))
         sys.exit(1)
+        
+
+def clean_ignored_docstrings(lines):
+    end_pattern = re.compile(r'("""\s*#\s*blocklint:.*pragma$|\'\'\'\s*#\s*blocklint:.*pragma$)')
```

Comment:
I was thinking of a multi-line regex, but that'd probably be overly complicated even before looking at nested strings!  I think this is a good solution.

Two points:
- Maybe `r'("""|\'\'\')\s*#\s*blocklint:.*pragma$'` would be easier to read and maintain?
- Are there spaces allowed after pragma?

Reply:
Yah, I was having trouble getting the regex right and just stopped when it worked. Will condense.

Right now it would not match on trailing spaces. Probably a good idea.